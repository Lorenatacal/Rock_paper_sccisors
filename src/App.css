.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 40s linear;
  height: 20vmin;
  pointer-events: none;
}

.App-header {
  background-color:white;
  margin-top: 2%;
  align-items: center;
  font-size: 90%;
  color: black;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
