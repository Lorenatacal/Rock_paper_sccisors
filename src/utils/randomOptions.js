let options = ["rock", "paper", "scissorss"];

export default function randomOptions() {
  const index = Math.random() * options.length;
  return options[Math.floor(index)];
}
