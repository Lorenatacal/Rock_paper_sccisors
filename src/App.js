import React, { Component } from "react";
import Header from "./Header.js";
import UserScreen from "./UserScreen.js";
import ComputerScreen from "./ComputerScreen.js";
import randomOptions from "./utils/randomOptions.js";
import gameEngine from "./game-engine/gameEngine.js";
import styled from "styled-components";

const StyledWrapper = styled.div`
  margin: 1%;
  text-align: center;
`;
const StyledResult = styled.div`
  margin-left: 1%;
  margin-right: 1%;
  background-image: linear-gradient(to right, orangered, rgb(255, 69, 0, 0.7));
  color: black;
`;
const StyledUserValue = styled.p`
  width: 38%;
  float: left;
`;
const StyledComputerValue = styled.p`
  width: 38%;
  float: right;
`;
const StyledWinner = styled.p`
  text-align: center;
  margin-top: -1%;
`;

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userValue: "",
      computerValue: "",
      userScore: 0,
      computerScore: 0
    };
  }

  handleClick = e => {
    let { userScore, computerScore } = gameEngine(
      this.state.userValue,
      this.state.computerValue,
      this.state.userScore,
      this.state.computerScore
    );
    this.setState({
      userValue: e.target.name,
      computerValue: randomOptions(),
      userScore,
      computerScore
    });
  };

  render() {
    let { winner, userScore, computerScore } = gameEngine(
      this.state.userValue,
      this.state.computerValue,
      this.state.userScore,
      this.state.computerScore
    );

    return (
      <StyledWrapper>
        <Header />
        <StyledResult>
          <StyledUserValue>{this.state.userValue}</StyledUserValue>
          <StyledComputerValue>{this.state.computerValue}</StyledComputerValue>
          <StyledWinner>{winner}</StyledWinner>
          <p>
            {userScore}-{computerScore}
          </p>
        </StyledResult>
        <UserScreen action={this.handleClick} />
        <ComputerScreen />
      </StyledWrapper>
    );
  }
}

export default App;
