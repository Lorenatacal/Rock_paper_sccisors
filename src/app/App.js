import React, { Component } from "react";
import Header from "../components/header/Header.js";
import UserScreen from "../components/userScreen/UserScreen.js";
import ComputerScreen from "../components/computerScreen/ComputerScreen.js";
import randomOptions from "../utils/randomOptions.js";
import gameEngine from "../game-engine/gameEngine.js";
import styled from "styled-components";

const StyledWrapper = styled.div`
  margin: 1%;
  text-align: center;
`;
const StyledResult = styled.section`
  width: 98%;
  background-image: linear-gradient(to right, orangered, rgb(255, 69, 0, 0.7));
  color: black;
  display: inline-flex;
  justify-content: space-around;
`;
const StyledPlayers = styled.section``;
const StyledUserValue = styled.span``;
const StyledComputerValue = styled.span``;
const StyledWinner = styled.span``;
const StyledScore = styled.span``;

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userValue: "",
      computerValue: "",
      userScore: 0,
      computerScore: 0
    };
  }

  handleClick = e => {
    let { userScore, computerScore } = gameEngine(
      this.state.userValue,
      this.state.computerValue,
      this.state.userScore,
      this.state.computerScore
    );

    this.setState({
      userValue: e.target.name,
      computerValue: randomOptions(),
      userScore,
      computerScore
    });
  };

  render() {
    let { winner, userScore, computerScore } = gameEngine(
      this.state.userValue,
      this.state.computerValue,
      this.state.userScore,
      this.state.computerScore
    );

    return (
      <StyledWrapper>
        <Header />
        <StyledResult>
          <StyledUserValue>{this.state.userValue}</StyledUserValue>
          <StyledWinner>{winner}</StyledWinner>
          <StyledScore>
            {userScore}-{computerScore}
          </StyledScore>
          <StyledComputerValue>{this.state.computerValue}</StyledComputerValue>
        </StyledResult>
        <StyledPlayers>
          <UserScreen action={this.handleClick} />
          <ComputerScreen />
        </StyledPlayers>
      </StyledWrapper>
    );
  }
}

export default App;
