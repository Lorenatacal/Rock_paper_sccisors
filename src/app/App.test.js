import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import randomOptions from "../utils/randomOptions.js";

Enzyme.configure({ adapter: new Adapter() });
jest.mock("../utils/randomOptions.js");

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
test("App should render correctlyt", () => {
  const wrapper = Enzyme.shallow(<App />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
test("App should render correctly on handleClick", () => {
  const wrapper = Enzyme.shallow(<App />);
  const e = {
    target: {
      name: "scissors"
    }
  };
  const userValue = "scissors";
  randomOptions.mockReturnValue("paper");
  const computerValue = randomOptions();

  wrapper.instance().handleClick(e);
  wrapper.update();
  console.log(wrapper.state().userScore, "d");

  expect(wrapper.state().userValue).toEqual("scissors");
  expect(wrapper.state().computerValue).toEqual("paper");
  expect(wrapper.state().computerScore).toEqual(0);
  expect(toJson(wrapper)).toMatchSnapshot();
});
