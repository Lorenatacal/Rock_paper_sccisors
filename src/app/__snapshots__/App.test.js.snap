// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly on handleClick 1`] = `
<styled.div>
  <Header />
  <styled.section>
    <styled.span>
      scissors
    </styled.span>
    <styled.span>
      User won
    </styled.span>
    <styled.span>
      1
      -
      0
    </styled.span>
    <styled.span>
      paper
    </styled.span>
  </styled.section>
  <styled.section>
    <UserScreen
      action={[Function]}
    />
    <ComputerScreen />
  </styled.section>
</styled.div>
`;

exports[`App should render correctly on handleClick 2`] = `
<styled.div>
  <Header />
  <styled.section>
    <styled.span>
      scissors
    </styled.span>
    <styled.span>
      User won
    </styled.span>
    <styled.span>
      2
      -
      0
    </styled.span>
    <styled.span>
      paper
    </styled.span>
  </styled.section>
  <styled.section>
    <UserScreen
      action={[Function]}
    />
    <ComputerScreen />
  </styled.section>
</styled.div>
`;

exports[`App should render correctlyt 1`] = `
<styled.div>
  <Header />
  <styled.section>
    <styled.span />
    <styled.span>
      Draw
    </styled.span>
    <styled.span>
      0
      -
      0
    </styled.span>
    <styled.span />
  </styled.section>
  <styled.section>
    <UserScreen
      action={[Function]}
    />
    <ComputerScreen />
  </styled.section>
</styled.div>
`;
